crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -130,21 +134,21 @@ impl ModuleSubscriptionActor {
     ) -> Result<(), DBError> {
         self.remove_subscriber(sender.id);
         let auth = AuthCtx::new(self.owner_identity, sender.id.identity);
-
+        let mut subscriptions = self.subscriptions.write_arc();
```

Comment:
is the write_arc() necessary here over just write()? and in general I'd say probably drop the lock as soon as possible barring interactions with external things that need to logically happen under the mutex. So, maybe before the send_message() call at the end? though maybe that was happening under the lock before anyway.

Reply:
Good point. Also, There was no lock here before.